src/platform/native/Core.zig

```diff
@@ -815,6 +826,7 @@ pub fn setVSync(self: *Core, mode: VSyncMode) void {
         .double => .fifo,
         .triple => .mailbox,
     };
+    if (mode != .triple) self.frame.target = 0;
```

Comment:
There are issues with this:

1. `setVSync` may be called from any thread, but `self.frame` is not guarded and may only be used from the render thread. i.e. `self.frame` may only be modified in `appUpdateThread` in this file.
2. If someone changes the vsync mode to e.g. `.none` or `.double`, you are removing the frame rate limit.. but if they set it back to `.triple` you aren't restoring the monitor limit.
3. If someone uses `core.setFrameRateLimit`, we need to take that into account in save/restore logic. i.e. if they set a limit of their own, `setVSync(.triple)` should override their limit unless they call `core.setFrameRateLimit()` again afterwards. But once they change the vsync mode back to say `.double` or `.none`, it should restore their frame rate limit.